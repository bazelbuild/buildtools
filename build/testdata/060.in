strings = [
  # empty
  '',
  "",
  '''''',
  """""",

  # ordinary
  'foo',
  "foo",
  '''foo''',
  """foo""",

  # contain quotes of different style
  '"foo"',
  "'foo'",
  '''"""foo"""''',
  """'''foo'''""",

  # contain quotes of the same style
  '\'foo\'',
  "\"foo\"",
  '''\'\'\'foo\'\'\'''',
  """\"\"\"foo\"\"\"""",

  # contain quotes of both styles
  '\'"foo"\'',
  "'\"foo\"'",
  '''\'\'\'"""foo"""\'\'\'''',
  """'''\"\"\"foo\"\"\"'''""",

  # raw strings
  r'\foo',
  r'"\foo',
  r"\foo",
  r"'\foo",
  r'''\foo''',
  r'''"""\foo''',
  r"""\foo""",
  r"""'''\foo""",
  r"\a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x43\y\z\0\1\2\3\4\5\6\7\8\9\\n",

  # contain incorrect escape sequences
  "\c'",
  '\c"',
  """\c'""",
  '''\c"''',
  "\a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x43\y\z\0\1\2\3\4\5\6\7\8\9",

  # correct escape sequences
  """\a\b\f\n\r\t\v\007\010\
  """,
  '''\a\b\f\n\r\t\v\007\010\
  ''',
  """ aa\\bb\n \
  """,
  "\000\111\222",
]
