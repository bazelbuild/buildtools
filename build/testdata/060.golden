strings = [
    # empty
    "",
    "",
    """""",
    """""",

    # ordinary
    "foo",
    "foo",
    """foo""",
    """foo""",

    # contain quotes of different style
    '"foo"',
    "'foo'",
    '''"""foo"""''',
    """'''foo'''""",

    # contain quotes of the same style
    "'foo'",
    "\"foo\"",
    """'''foo'''""",
    """\"\"\"foo\"\"\"""",

    # contain quotes of both styles
    '\'"foo"\'',
    "'\"foo\"'",
    '''\'\'\'"""foo"""\'\'\'''',
    """'''\"\"\"foo\"\"\"'''""",

    # raw strings
    r"\foo",
    r'"\foo',
    r"\foo",
    r"'\foo",
    r"""\foo""",
    r'''"""\foo''',
    r"""\foo""",
    r"""'''\foo""",
    r"\a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x43\y\z\0\1\2\3\4\5\6\7\8\9\\n",

    # contain incorrect escape sequences
    "\\c'",
    "\\c\"",
    """\\c'""",
    """\\c\"""",
    "\a\b\\c\\d\\e\f\\g\\h\\i\\j\\k\\l\\m\n\\o\\p\\q\r\\s\t\\u\v\\wC\\y\\z\000\001\002\003\004\005\006\a\\8\\9",

    # correct escape sequences
    """\a\b\f\n\r\t\v\007\010\
  """,
    """\a\b\f
\r\t\v\a\b  """,
    """ aa\\bb\n \
  """,
    "\000\111\222",
]
